src/app/AlwaysOn.Shared/Services/CosmosDbService.cs

```diff
@@ -178,204 +181,123 @@ public async Task DeleteItemAsync<T>(string objectId, string partitionKey)
 
         public async Task<CatalogItem> GetCatalogItemByIdAsync(Guid itemId)
         {
-            string partitionKey = itemId.ToString();
-            var startTime = DateTime.UtcNow;
+            var requestOptions = AppInsightsRequestHandler.CreateOptionsWithOperation<ItemRequestOptions>(nameof(GetCatalogItemByIdAsync), _dbClient.Endpoint.Host);
+
             ResponseMessage responseMessage = null;
-            CosmosDiagnostics diagnostics = null;
-            var success = false;
             try
             {
                 // Read the item as a stream for higher performance.
                 // See: https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Exceptions.md#stream-api
-                responseMessage = await _catalogItemsContainer.ReadItemStreamAsync(
-                    partitionKey: new PartitionKey(partitionKey),
-                    id: itemId.ToString());
-                diagnostics = responseMessage.Diagnostics;
+                responseMessage = await _catalogItemsContainer.ReadItemStreamAsync(itemId.ToString(), new PartitionKey(itemId.ToString()), requestOptions);
 
                 // Item stream operations do not throw exceptions for better performance
                 if (responseMessage.IsSuccessStatusCode)
                 {
-                    var item = await JsonSerializer.DeserializeAsync<CatalogItem>(responseMessage.Content, Globals.JsonSerializerOptions);
-                    success = true;
-                    return item;
+                    try
+                    {
+                        var item = await JsonSerializer.DeserializeAsync<CatalogItem>(responseMessage.Content, Globals.JsonSerializerOptions);
+                        return item;
```

Comment:
why not directly return on the previous line?

Reply:
Good catch - I managed to do it elsewhere, but missed here :D